### ISSUE TITLE
Weird waveform rendering.

### ISSUE THREAD

**Issue by M0JXD** [Stazed/non-timeline-xt#6]
Hi again! Loving the project.

I've had an issue with a weird waveform rendering - one particular recording doesn't match the real waveform in my project. Here it is compared to Audacity:

![waveform-comparison](https://github.com/user-attachments/assets/dd0b42e3-9245-4648-afe7-0bb6e7f73055)

You can see the render sort of repeats itself. Aside from the first quiet part no other parts are silent, and the playback is correct.

I've checked and this is the same in the original non-timeline (using the deb from KXStudio) when I open with the project files. If I import the file on a separate track or in a new project there are no rendering issues so I imagine it's something weird with the journal/history? It was wrong from the first recording of the track (which was done with non-xt).

Here is my (compressed) RaySession folder if it helps. Excuse the playing, it's just a rough first go...

https://drive.google.com/file/d/1vXOADQGzsHP_75_2HDtgckN0xh-urzJt/view?usp=sharing

**Comment by Stazed** [comment]
O.K. this is a weird one alright. It seems to be caused by a bad .peak file. I was able to fix by simply deleting the related peak file 173332614.wav.peak in the sources folder. The peak file will be automatically re-generated.

Not sure what caused the peak file issue, anything unusual about how it was recorded? Ever notice this happening in other cases?

**Comment by M0JXD** [comment]
Not really, just guitar -> guitar modeller -> USB -> JACK -> Non-Timeline-XT. The backing track was playing with it from Non-Timeline-XT -> Non-Mixer-XT to JACK's output, and the modeller provides a direct out. JACK was at 1024 samples (20ms latency, really ought to change that) and you've seen my PC specs so I doubt I ran out out of processing power and messed up generating the peak file - didn't see any xruns.

The second double-track was done in the same way with no issues? I will try and recreate it.

**Comment by M0JXD** [comment]
I've tried a fair few times to get this to repeat but I can't. I didn't realise before that the peak file is 1.1GB so something clearly went wrong.

The only thing that's changed recently is that I've had to "downgrade" the monitor/TV on my desk just two days ago, it's much older and certainly worse than my previous one. The display driver is doing odd things and just before this recording it had an problem that I had to fix in Nvidia's settings. Maybe it's upsetting the audio driver at ALSA's level as well, even though I do not use it as an audio device (it's available via the Nvidia's HDMI) an upset in ALSA could upset JACK too, and cascade the problem to Non-XT. Non-Mixer's window was on that monitor. I didn't think the two could be linked but maybe that's the case.

If it happens again I know the fix at least! And I'll let you know if so as well.

**Comment by Stazed** [comment]
Yeah, I noticed the file size as well, and it is kind of scary. Would really like to know what caused this.

**Comment by M0JXD** [comment]
FYI on this issue, I found this:
https://web.archive.org/web/20201123000254/https://github.com/original-male/non/issues/198

I've been using Non-XT quite a lot and it's never occurred again. Very weird one!
